quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -183,16 +185,19 @@ fn delete_source_handler(
         .and(warp::delete())
         .and(with_arg(index_service))
         .then(delete_source)
-        .map(format_response)
 }
 
 async fn delete_source(
     index_id: String,
     source_id: String,
     index_service: Arc<IndexService>,
-) -> Result<(), IndexServiceError> {
+) -> impl warp::Reply {
     info!(index_id = %index_id, source_id = %source_id, "delete-source");
-    index_service.delete_source(&index_id, &source_id).await
+    let res = index_service.delete_source(&index_id, &source_id).await;
```

Comment:
It works... but it's really not a good solution, unfortunately. We have other DELETE endpoints that may return just an empty body. And an empty body is not a valid JSON.

Having something like `Format::make_rest_reply_non_serializable_error` specific to DELETE requests.